client/src/utils/responsiveness.ts

```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react'
+
+/**
+ * Given a CSS media query, returns whether the media query is matched, e.g.
+ *
+ * const isDesktopWidth = useMediaQuery('(min-width: 64em)')
+ *
+ * Listens for updates, e.g. in response to window resizing.
+ */
+export function useMediaQuery(query: string): boolean {
+  const [isMatched, setIsMatched] = useState(false)
+
+  useEffect(() => {
+    const matchMedia = window.matchMedia(query)
```

Comment:
CSS media queries can take you far, but sometimes you need to control a prop/styles in a third-party library or tweak something more complex that can't be accomplished by hiding/unhiding elements.

In the past, when I've inevitably come across this need to access the screen width in JS, I've used `window.innerWidth` plus a debounced window resize listener. This time around, I discovered [`window.matchMedia`](https://developer.mozilla.org/en-US/docs/Web/API/Window/matchMedia), which as noted in the PR description, is magic 🪄 and basically brings the power of CSS media queries to JS! It's supported in all modern browsers (and non-modern browsers, too, like IE lol).

Relevant article: https://webdevetc.com/blog/matchmedia-events-for-window-resizes